### PR TITLE
VPN Client Protocols for gateways

### PR DESCRIPTION
**Pull request by henry416**
## Description

Network API change. Enable IkeV2 and SSTP VPN client protocols for virtual network gateways.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by henry416** [comment]
@markcowl Please review

**Comment by henry416** [comment]
@markcowl Is this being reviewed?

**Comment by henry416** [comment]
@cormacpayne Requesting alternate reviewer as it appears mark of OOF

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayTests.cs [hunk 1]
```diff
@@ -57,5 +57,11 @@ public void VirtualNetworkGatewayRouteApiTest()
         {
             NetworkResourcesController.NewInstance.RunPsTest("Test-VirtualNetworkGatewayBgpRouteApi");
         }
+
+        [Fact]
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayTests.cs]: @henry416 please make sure that this is a `CheckIn` test so it runs during the regular build

**Comment by cormacpayne** [comment]
@henry416 in addition to the comments above, please make sure to pull the latest changes from the  `preview` branch since we had release 4.3.1 go out last week

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayTests.cs (hunk 1 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayTests.cs]: Category.CheckIn is already in place

Comment (henry416) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayTests.cs]: Never mind - read wrong section - fixed now

**Comment by henry416** [comment]
@cormacpayne I have added checkin test, is there any other pending?

**Comment by henry416** [comment]
@cormacpayne The added test has passed - not sure what is failing the build

**Comment by cormacpayne** [comment]
@henry416 the build is failing because the CredScan tool is finding potential keys/passwords/secrets in your files, specifically the `VirtualNetworkGatewayIkeV2Test.json` file. You can find the matches under `Build Artifacts > src > Package > credscan-matches.tsv`, and you can find information on how to suppress these matches if they are not actually keys/passwords/secrets at aka.ms/credscan. Ping me offline if these turn out to be actual secrets.

**Comment by henry416** [comment]
@cormacpayne Test has been updated with suppression

**Comment by henry416** [comment]
@cormacpayne please review

**Comment by henry416** [comment]
related PR: https://github.com/Azure/azure-powershell/pull/4529
Note that this current PR is not dependent on referenced one.

File: src/ResourceManager/Network/ChangeLog.md [hunk 2]
```diff
@@ -32,6 +32,12 @@
     - Updated Add-AzureRMLoadBalancerRuleConfig: Added optional parameter DisableOutboundSNAT
     - Updated Set-AzureRMLoadBalancerRuleConfig: Added optional parameter DisableOutboundSNAT
 
+## Version 4.3.2
+* New-AzureRmVirtualNetworkGateway: Added IkeV2 P2S support. New optional parameter added
```
Comment (cormacpayne) [inline src/ResourceManager/Network/ChangeLog.md]: @henry416 please remove the `## Version 4.3.2` header and move the contents you are adding to the `## Current Release` section

**Pull request merged**